### PR TITLE
Compile always and generate package.json

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

**Comment by bergundy** [comment]
@vogre PTAL

File: src/output.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  },
+  include: [
+    'src/*.ts',
+  ],
+  exclude: [
+    'node_modules',
+  ],
+};
+
+const spawn = (command: string, args?: ReadonlyArray<string>, options?: SpawnOptions) =>
```
Comment (vogre) [inline src/output.ts]: This spawn wrapper does not check if the command failed?

File: src/output.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  public async write(
+    name: string,
+    version: string,
+    { code, pkg: basePackage }: GeneratedCode,
+    role: Role
+  ) {
+    try {
+      await mkdir(path.join(this.genPath, 'src'));
```
Comment (vogre) [inline src/output.ts]: This might have been simpler by using `mkdirp` package.

File: src/output.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  compilerOptions: {
+    lib: ['es2017', 'esnext'],
+    target: 'es2017',
+    module: 'commonjs',
+    moduleResolution: 'node',
+    emitDecoratorMetadata: true,
+    checkJs: false,
+    allowJs: false,
+    experimentalDecorators: true,
+    downlevelIteration: true,
```
Comment (vogre) [inline src/output.ts]: No need for `downlevelIteration` with es2017, also `checkJs` and `allowJs` are false by default.

File: src/lib/common.ts [hunk 4]
```diff
@@ -10,10 +10,10 @@ export function coerceWithSchema(schema: any, value: any, defsSchema = {}): any
     if (Array.isArray(schema.items)) {
       return zip(schema.items, value).map(([s, v]) => coerceWithSchema(s, v, defsSchema));
     }
-    return value.map((v) => coerceWithSchema(schema.items, v, defsSchema));
+    return value.map((v: any) => coerceWithSchema(schema.items, v, defsSchema));
```
Comment (vogre) [inline src/lib/common.ts]: Types are good!

File: src/index.ts [hunk 5]
```diff
@@ -1,44 +1,83 @@
...
+  const clientOnly = {
+    dependencies: {
+      request: '^2.88.0',
+      'request-promise-native': '^1.0.5',
+    },
+    devDependencies: {
+      '@types/request': '^2.48.1',
+      '@types/request-promise-native': '^1.0.15',
+    },
+    // TODO: decide whether to include this, maybe as peerDependencies
```
Comment (vogre) [inline src/index.ts]: Obsolete TODO?

File: src/output.ts (hunk 2 above)
Comment (bergundy) [reply src/output.ts]: Yeah, but this is acceptable IMO

File: src/index.ts (hunk 5 above)
Comment (bergundy) [reply src/index.ts]: As discussed F2F made @types full on dependencies

File: src/output.ts (hunk 3 above)
Comment (bergundy) [reply src/output.ts]: OK

**Pull request merged**